magento/magento2
# Fix issue with no results on other than first search pages when using elasticsearch

### Description (*)

Pull request introduces the changes in search results applier in elastic search module which are resetting current page and page size on collection.
It is needed because elastic search applies condition on the collection with a particular set of entity ids, w/o change introduced by PR there are no results on search pages other than first one because there is a limit and offset applied on select query.

### Fixed Issues (if relevant)

1. magento/magento2#25378: No results on other than first search pages when using elasticsearch

### Manual testing scenarios (*)

1. Need to have configured elastic search as search engine.
2. Magento 2.3.3 EE but will occur on CE as well.
3. Products which will be found using some phrase and their count is higher than the configured page size.
4. Search using this phrase, on the bottom on search page pagination will be rendered, if we click on second, third, etc no result message will appear because the query will look like the following:
`(e.entity_id` IN (8357, 3524, 18671, 8360, 8390, 3971, 11213, 3527, 8387, 11393)) ORDER BY FIELD(e.entity_id,8357,3524,18671,8360,8390,3971,11213,3527,8387,11393) ASC
 LIMIT 10 OFFSET 10`

### Questions or comments


### Contribution checklist (*)
